### PR TITLE
[BugFix] Disable change replicate_storage into true for the table with GIN

### PR DESCRIPTION
**Pull request by srlch**
Currently Gin does not support replicate storage, but we can use ALTER stmt to change a FALSE one into a TRUE one even the table has GIN.

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by trueeyu — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AlterTableClauseVisitor.java [hunk 1]
```diff
@@ -250,6 +250,20 @@ public Void visitModifyTablePropertiesClause(ModifyTablePropertiesClause clause,
...
+                OlapTable olapTable = (OlapTable) table;
+                boolean hasGIN = false;
+                for (Index index : olapTable.getIndexes()) {
+                    if (index.getIndexType() == IndexDef.IndexType.GIN) {
+                        hasGIN = true;
+                        break;
+                    }
+                }
+                if (properties.get(PropertyAnalyzer.PROPERTIES_REPLICATED_STORAGE).equalsIgnoreCase("true") && hasGIN) {
+                    ErrorReport.reportSemanticException(ErrorCode.ERR_COMMON_ERROR,
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AlterTableClauseVisitor.java:263]: Put the error message into ErrorCode.java

Comment (srlch) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AlterTableClauseVisitor.java]: Fixed

**Review by jaogoy — DISMISSED** [review]
For msgs

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 2]
```diff
@@ -286,6 +286,8 @@ public enum ErrorCode {
             "The default storage volume does not exist. " +
                     "A default storage volume can be created by following these steps: " +
                     "1. Create a storage volume. 2. Set the storage volume as default"),
+    ERR_GIN_REPLICATED_STORAGE_NOT_SUPPORTED(5507, new byte[] {'0', 'A', '0', '0', '0'},
+                "Can not use replicated storage as TRUE when the table has GIN"),
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:290]:
```suggestion
                "Can not enable replicated storage when the table has GIN"),
```

Comment (srlch) [reply fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java]: done

**Review by wyb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by jaogoy — APPROVED** [review]

**Pull request merged**